### PR TITLE
Strip out old XCMP primitives

### PR DESCRIPTION
**Pull request by expenses**
See https://github.com/paritytech/polkadot/issues/793.

### CODE REVIEW

**Comment by expenses** [comment]
@rphmeier should be ready for review, but some of the tests are failing.

File: network/src/gossip/message_routing.rs [hunk 1]
```diff
@@ -139,18 +139,11 @@ impl View {
 		match self.expected_queues.get_mut(&ostensible_topic) {
 			None => (GossipValidationResult::Discard, super::cost::UNNEEDED_ICMP_MESSAGES),
 			Some(&mut (_, ref mut known)) => {
-				if !messages.queue_root_is_correct() {
```
Comment (rphmeier) [inline network/src/gossip/message_routing.rs]: you can probably just kill this file and the relevant gossip message variants

File: runtime/common/src/parachains.rs [hunk 2]
```diff
@@ -618,7 +618,7 @@ impl<T: Trait> Module<T> {
 	/// invoked off-chain.
 	///
 	/// Yields a structure containing all unrouted ingress to the parachain.
-	pub fn ingress(to: ParaId, since: Option<T::BlockNumber>) -> Option<Vec<(T::BlockNumber, BlockIngressRoots)>> {
+	pub fn ingress(to: ParaId, since: Option<T::BlockNumber>) -> Option<Vec<(T::BlockNumber, Vec<(ParaId, Hash)>)>> {
```
Comment (rphmeier) [inline runtime/common/src/parachains.rs]: you can just remove this function

Comment (rphmeier) [reply runtime/common/src/parachains.rs]: (watermarks, `UnroutedIngress` storage and logic using them can be removed as well)

Comment (rphmeier) [reply runtime/common/src/parachains.rs]: along with the relevant `ParachainHost` functions

File: availability-store/src/store.rs [hunk 3]
```diff
@@ -238,7 +226,7 @@ impl Store {
 	/// Add a set of chunks.
 	///
 	/// The same as `add_erasure_chunk` but adds a set of chunks in one atomic transaction.
-	/// Checks that all chunks have the same `relay_parent`, `block_data_hash` and `parachain_id` fields.
+	/// Checks that all chunks have the same `relay_parent`, `block_hash` and `parachain_id` fields.
```
Comment (rphmeier) [inline availability-store/src/store.rs]: Maybe an accidental change? The `block_data` is a Polkadot term for the witness data accompanying a `CandidateReceipt`, which proves the blocks validity. This is not the same as the parachain block itself.

Comment (expenses) [reply availability-store/src/store.rs]: Fixed.

File: availability-store/src/lib.rs [hunk 4]
```diff
@@ -124,8 +123,8 @@ pub struct Data {
 	pub relay_parent: Hash,
 	/// The parachain index for this candidate.
 	pub parachain_id: ParaId,
-	/// The block itself.
-	pub block: PoVBlock,
+	/// Block data.
+    pub block_data: BlockData,
```
Comment (rphmeier) [inline availability-store/src/lib.rs]: spaces

File: availability-store/src/worker.rs [hunk 5]
```diff
@@ -90,7 +90,7 @@ pub(crate) struct ParachainBlocks {
 	/// The relay parent of the block these parachain blocks belong to.
 	pub relay_parent: Hash,
 	/// The blocks themselves.
-	pub blocks: Vec<(CandidateReceipt, Option<PoVBlock>)>,
+	pub blocks: Vec<(CandidateReceipt, Option<BlockData>)>,
```
Comment (rphmeier) [inline availability-store/src/worker.rs]: Why replace this with `BlockData`? Shouldn't it be `PoVBlock`?

File: primitives/src/parachain.rs [hunk 6]
```diff
@@ -580,12 +460,6 @@ sp_api::decl_runtime_apis! {
 		fn parachain_status(id: Id) -> Option<Status>;
 		/// Get the given parachain's head code blob.
 		fn parachain_code(id: Id) -> Option<Vec<u8>>;
-		/// Get all the unrouted ingress roots at the given block that
-		/// are targeting the given parachain.
-		///
-		/// If `since` is provided, only messages since (including those in) that block
-		/// will be included.
-		fn ingress(to: Id, since: Option<BlockNumber>) -> Option<StructuredUnroutedIngress>;
```
Comment (rphmeier) [inline primitives/src/parachain.rs]: cc @bkchr will this brick existing chains without a runtime API version upgrade?

File: runtime/common/src/parachains.rs [hunk 7]
```diff
@@ -147,10 +141,6 @@ const EMPTY_TRIE_ROOT: [u8; 32] = [
 
 /// Total number of individual messages allowed in the parachain -> relay-chain message queue.
 const MAX_QUEUE_COUNT: usize = 100;
-/// Total size of messages allowed in the parachain -> relay-chain message queue before which no
-/// further messages may be added to it. If it exceeds this then the queue may contain only a
-/// single message.
-const WATERMARK_QUEUE_SIZE: usize = 20000;
```
Comment (rphmeier) [inline runtime/common/src/parachains.rs]: actually, this should be restored along with the logic in `check_upward_messages` which used it.

File: runtime/common/src/parachains.rs [hunk 8]
```diff
@@ -426,106 +368,20 @@ impl<T: Trait> Module<T> {
...
-				}
-			}
-
-			// place our egress root to `to` into the ingress table for (now, `to`).
-			for &(to, root) in &head.candidate.egress_queue_roots {
-				ingress_update.entry(to).or_insert_with(Vec::new).push((id, root));
-			}
-
-			// Queue up upwards messages (from parachains to relay chain).
-			Self::queue_upward_messages(
```
Comment (rphmeier) [inline runtime/common/src/parachains.rs]: This call to `queue_upward_messages` is still relevant, but it's pretty much the only thing that should remain.

File: runtime/common/src/parachains.rs [hunk 9]
```diff
@@ -426,106 +368,20 @@ impl<T: Trait> Module<T> {
...
-				id,
-				&head.candidate.upward_messages,
-				&mut ordered_needs_dispatch,
-			);
-		}
-
-		NeedsDispatch::put(ordered_needs_dispatch);
```
Comment (rphmeier) [inline runtime/common/src/parachains.rs]: This should be reinstated as well

File: runtime/common/src/parachains.rs [hunk 10]
```diff
@@ -426,106 +368,20 @@ impl<T: Trait> Module<T> {
...
-			<UnroutedIngress<T>>::insert((now, to), ingress_roots);
-		}
-	}
-
-	/// Place any new upward messages into our queue for later dispatch.
-	///
-	/// `ordered_needs_dispatch` is mutated to ensure it reflects the new value of
-	/// `RelayDispatchQueueSize`. It is up to the caller to guarantee that it gets written into
-	/// storage after this call.
-	fn queue_upward_messages(
```
Comment (rphmeier) [inline runtime/common/src/parachains.rs]: This should be reinstated as it's parachain -> relay-chain messaging, not parachain -> parachain

File: runtime/common/src/parachains.rs [hunk 11]
```diff
@@ -426,106 +368,20 @@ impl<T: Trait> Module<T> {
...
-			} else {
-				sp_runtime::print("ordered_needs_dispatch contains id?!");
-			}
-		}
-	}
-
 	/// Simple FIFO dispatcher.
 	fn dispatch_upward_messages(
 		max_queue_count: usize,
-		watermark_queue_size: usize,
```
Comment (rphmeier) [inline runtime/common/src/parachains.rs]: should reinstate this and the logic that used it

File: runtime/common/src/parachains.rs [hunk 12]
```diff
@@ -1351,253 +1178,6 @@ mod tests {
...
 	}
 
-	fn queue_upward_messages(id: ParaId, upward_messages: &[UpwardMessage]) {
-		NeedsDispatch::mutate(|nd|
-			Parachains::queue_upward_messages(id, upward_messages, nd)
-		);
-	}
-
-	#[test]
-	fn check_dispatch_upward_works() {
```
Comment (rphmeier) [inline runtime/common/src/parachains.rs]: this and other upward message tests should come back

**Comment by rphmeier** [comment]
Looks like the egress queue tests in the Polkadot runtime need to be removed. Otherwise looks good.

File: primitives/src/parachain.rs (hunk 6 above)
Comment (bkchr) [reply primitives/src/parachain.rs]: Looks like it was only called in conjunction with Parachain validation? So, as we don't have any Parachain, there should be no problem.

File: runtime/common/src/parachains.rs [hunk 13]
```diff
@@ -301,15 +281,9 @@ decl_module! {
 				}
 
 				let para_blocks = Self::check_candidates(&heads, &active_parachains)?;
-				let current_number = <system::Module<T>>::block_number();
 
 				<attestations::Module<T>>::note_included(&heads, para_blocks);
 
-				Self::update_routing(
```
Comment (rphmeier) [inline runtime/common/src/parachains.rs]: the last commit reinstated `update_routing` but the call site is still removed

File: runtime/common/src/parachains.rs [hunk 14]
```diff
@@ -169,20 +163,7 @@ decl_storage! {
...
-		/// Unrouted ingress. Maps (BlockNumber, to_chain) pairs to [(from_chain, egress_root)].
-		///
-		/// There may be an entry under (i, p) in this map for every i between the parachain's
-		/// watermark and the current block.
-		pub UnroutedIngress:
-			map hasher(blake2_256) (T::BlockNumber, ParaId) => Option<Vec<(ParaId, Hash)>>;
-
-		/// Messages ready to be dispatched onto the relay chain. It is subject to
-		/// `MAX_MESSAGE_COUNT` and `WATERMARK_MESSAGE_SIZE`.
+		/// Messages ready to be dispatched onto the relay chain.
```
Comment (rphmeier) [inline runtime/common/src/parachains.rs]: why change docs?

**Pull request merged**